### PR TITLE
[azure] [fix] Fixed collecting network resources + added tests

### PR DESCRIPTION
**Pull request by 1101-1**
# Description

1. Fixed Checksum Bug: Resolved an issue related to checksum calculation for resources. The issue was causing duplicate checksums for certain resource types. This problem has been fixed by ensuring that keys used for checksum calculation are distinct and unique for each resource.

2. Fixed Resource Collection: Refactored and improved the resource collection process. This change enhances the collection of resources, ensuring they are correctly identified and processed.

3. Added Tests: Included new test cases to validate the correctness of the resource collection.

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [x] Fix collecting network resources
- [x] Fix bug with duplicate checksum

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/azure/resoto_plugin_azure/resource/network.py [hunk 1]
```diff
@@ -3242,7 +3240,7 @@ class AzureExpressRoutePort(AzureResource):
         "ether_type": S("properties", "etherType"),
         "identity": S("identity") >> Bend(AzureManagedServiceIdentity.mapping),
         "links": S("properties", "links") >> ForallBend(AzureExpressRouteLink.mapping),
-        "mtu_string": S("properties", "mtu"),  # TODO: check if this string is actually an int?
+        "mtu_int": S("properties", "mtu") >> AsInt(),
```
Comment (aquamatthias) [inline plugins/azure/resoto_plugin_azure/resource/network.py:3243]:
```suggestion
        "mtu": S("properties", "mtu") >> AsInt(),
```

File: plugins/azure/resoto_plugin_azure/resource/network.py [hunk 2]
```diff
@@ -4448,14 +4446,18 @@ class AzureUsage(AzureResource):
     mapping: ClassVar[Dict[str, Bender]] = {
         "id": S("id"),
         "tags": S("tags", default={}),
-        "name": S("name"),
+        "name": K(None),
+        "usage_name": S("name") >> Bend(AzureUsageName.mapping),
         "ctime": K(None),
         "mtime": K(None),
         "atime": K(None),
```
Comment (aquamatthias) [inline plugins/azure/resoto_plugin_azure/resource/network.py:4453]: Can you please review all `K(None)` values to see if there is a value we can derive this value from? 
Ultimately, we should not have any of them in the code.

**Review by aquamatthias — COMMENTED** [review]
Thanks!

File: plugins/azure/resoto_plugin_azure/resource/network.py [hunk 3]
```diff
@@ -3491,7 +3489,7 @@ class AzureFirewallPolicy(AzureResource):
         >> Bend(AzureFirewallPolicyIntrusionDetection.mapping),
         "provisioning_state": S("properties", "provisioningState"),
         "rule_collection_groups": S("properties") >> S("ruleCollectionGroups", default=[]) >> ForallBend(S("id")),
-        "size_string": S("properties", "size"),  # TODO: check if this is actually an int?
+        "size_string": S("properties", "size"),
```
[the remark this answers is not in the document]
Comment (1101-1) [reply plugins/azure/resoto_plugin_azure/resource/network.py]: yeah

File: plugins/azure/resoto_plugin_azure/resource/network.py (hunk 2 above)
Comment (1101-1) [reply plugins/azure/resoto_plugin_azure/resource/network.py:4453]: Unfortunately, I did not find any `creation time`/`access time`/`modification time `in network resources. Azure does not provide it on network resources.

File: plugins/azure/resoto_plugin_azure/resource/network.py [hunk 4]
```diff
@@ -3512,7 +3510,7 @@ class AzureFirewallPolicy(AzureResource):
     intrusion_detection: Optional[AzureFirewallPolicyIntrusionDetection] = field(default=None, metadata={'description': 'Configuration for intrusion detection mode and rules.'})  # fmt: skip
     provisioning_state: Optional[str] = field(default=None, metadata={'description': 'The current provisioning state.'})  # fmt: skip
     rule_collection_groups: Optional[List[str]] = field(default=None, metadata={'description': 'List of references to FirewallPolicyRuleCollectionGroups.'})  # fmt: skip
-    size_string: Optional[str] = field(default=None, metadata={'description': 'A read-only string that represents the size of the FirewallPolicyPropertiesFormat in MB. (ex 0.5MB)'})  # fmt: skip
+    size_string: Optional[int] = field(default=None, metadata={'description': 'A read-only string that represents the size of the FirewallPolicyPropertiesFormat in MB. (ex 0.5MB)'})  # fmt: skip
```
Comment (aquamatthias) [inline plugins/azure/resoto_plugin_azure/resource/network.py:3513]: rename to size

File: plugins/azure/resoto_plugin_azure/resource/compute.py [hunk 5]
```diff
@@ -2150,7 +2150,7 @@ class AzureVirtualMachineNetworkInterfaceConfiguration:
 
 @define(eq=False, slots=False)
 class AzureNetworkProfile:
-    kind: ClassVar[str] = "azure_network_profile"
+    kind: ClassVar[str] = "azure_virtual_machine_network_profile"
```
Comment (aquamatthias) [inline plugins/azure/resoto_plugin_azure/resource/compute.py:2153]: Please revert. Azure Kubernetes Service (AKS) or Azure Container Instances (ACI) uses a network profile.

File: plugins/azure/resoto_plugin_azure/resource/network.py [hunk 6]
```diff
@@ -3491,7 +3489,7 @@ class AzureFirewallPolicy(AzureResource):
         >> Bend(AzureFirewallPolicyIntrusionDetection.mapping),
         "provisioning_state": S("properties", "provisioningState"),
         "rule_collection_groups": S("properties") >> S("ruleCollectionGroups", default=[]) >> ForallBend(S("id")),
-        "size_string": S("properties", "size"),  # TODO: check if this is actually an int?
+        "size_string": S("properties", "size") >> StringToUnitNumber("B"),
```
Comment (aquamatthias) [inline plugins/azure/resoto_plugin_azure/resource/network.py:3492]:
```suggestion
        "size_string": S("properties", "size") >> StringToUnitNumber("B", expected=int),
```

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**